### PR TITLE
asn 0.71.6 (new formula)

### PR DESCRIPTION
**Pull request by filippovitale**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This ASN lookup tool is a bash script from https://github.com/nitefood/asn for having a quick OSINT command line tool at disposal when investigating network data, which can come in handy in incident response scenarios as well.

### CODE REVIEW

File: Formula/asn.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Asn < Formula
+  desc "Organization lookup and server tool (ASN / IPv4 / IPv6 / Prefix / AS Path)"
+  homepage "https://github.com/nitefood/asn"
+  url "https://raw.githubusercontent.com/nitefood/asn/64e92533f57e9261d1d6588b4714a4183fc3c304/asn"
```
Comment (carlocab) [inline Formula/asn.rb:4]: This needs to be a versioned tarball. See https://docs.brew.sh/Acceptable-Formulae#stable-versions.

File: Formula/asn.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  version "0.71.6"
+  sha256 "277765b43a44ccf53df5d60437f7564f003d86d9c4e32e98b9eb0c8b409cf02e"
+  license "MIT"
+
+  livecheck do
+    skip "Cannot reliably check for new releases upstream"
+  end
+
+  bottle :unneeded
```
Comment (carlocab) [inline Formula/asn.rb:13]:
```suggestion
```
Let's remove this. See #75943.

File: Formula/asn.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  def install
+    bin.install "asn"
+  end
+
+  test do
+    assert_match version.to_s, shell_output(%Q|sed -n 's/^ASN_VERSION="\\(.*\\)"$/\\1/p' #{bin}/asn|)
```
Comment (carlocab) [inline Formula/asn.rb:20]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

This doesn't even seem like a test of `asn` so much as a test of `sed`.

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Comment by filippovitale** [comment]
Newer version of the same script is now tracked [in this PL](https://github.com/Homebrew/homebrew-core/pull/141539)

**Pull request closed without merging**